rx-preferences/src/main/java/com/f2prateek/rx/preferences/BackpressureBufferLastOperator.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.f2prateek.rx.preferences;
+
+import rx.Observable.Operator;
+import rx.Producer;
+import rx.Subscriber;
+
+/** An operator which keeps the last emitted instance when backpressure has been applied. */
+final class BackpressureBufferLastOperator<T> implements Operator<T, T> {
+  static final Operator<Object, Object> instance = new BackpressureBufferLastOperator<>();
+
+  static <T> Operator<T, T> instance() {
+    //noinspection unchecked
+    return (Operator<T, T>) instance;
+  }
+
+  private BackpressureBufferLastOperator() {
+  }
+
+  @Override public Subscriber<? super T> call(final Subscriber<? super T> child) {
+    BufferLastSubscriber<T> parent = new BufferLastSubscriber<>(child);
+    child.add(parent);
+    child.setProducer(parent.producer);
+    return parent;
+  }
+
+  static final class BufferLastSubscriber<T> extends Subscriber<T> {
+    private static final Object NONE = new Object();
+
+    private final Subscriber<? super T> child;
+
+    private volatile Object last = NONE; // Guarded by 'this'.
+    private volatile long requested; // Guarded by 'this'. Starts at zero.
+
+    final Producer producer = new Producer() {
+      @Override public void request(long n) {
+        if (n < 0) {
+          throw new IllegalArgumentException("requested " + n + " < 0");
```

Comment:
If you accidentally overflow a long (requests usually are `Long.MAX_VALUE` so you could blindly add 1 and get negative) or are just off-by-one (or more) on some subtraction code for request. But it's unlikely to ever happen in practice. Most of this file is copied from an RxJava operator.